bundles/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/handler/RokuHandler.java

```diff
@@ -191,21 +195,32 @@ private void refreshPlayerState() {
                             PLAY.equalsIgnoreCase(playerInfo.getState()) ? PlayPauseType.PLAY : PlayPauseType.PAUSE);
 
                     // Remove non-numeric from string, ie: ' ms'
-                    String position = playerInfo.getPosition().replaceAll(NON_DIGIT_PATTERN, EMPTY);
-                    if (!EMPTY.equals(position)) {
-                        updateState(TIME_ELAPSED,
-                                new QuantityType<>(Integer.parseInt(position) / 1000, API_SECONDS_UNIT));
+                    final String positionStr = playerInfo.getPosition().replaceAll(NON_DIGIT_PATTERN, EMPTY);
+                    int position = -1;
+                    if (!EMPTY.equals(positionStr)) {
+                        position = Integer.parseInt(positionStr) / 1000;
+                        updateState(TIME_ELAPSED, new QuantityType<>(position, API_SECONDS_UNIT));
                     } else {
                         updateState(TIME_ELAPSED, UnDefType.UNDEF);
                     }
 
-                    String duration = playerInfo.getDuration().replaceAll(NON_DIGIT_PATTERN, EMPTY);
-                    if (!EMPTY.equals(duration)) {
-                        updateState(TIME_TOTAL,
-                                new QuantityType<>(Integer.parseInt(duration) / 1000, API_SECONDS_UNIT));
+                    final String durationStr = playerInfo.getDuration().replaceAll(NON_DIGIT_PATTERN, EMPTY);
+                    int duration = -1;
+                    if (!EMPTY.equals(durationStr)) {
+                        duration = Integer.parseInt(durationStr) / 1000;
+                        updateState(TIME_TOTAL, new QuantityType<>(duration, API_SECONDS_UNIT));
                     } else {
                         updateState(TIME_TOTAL, UnDefType.UNDEF);
                     }
+
+                    if (position >= 0 && duration > 0) {
+                        updateState(END_TIME, new DateTimeType(ZonedDateTime.now().plusSeconds(duration - position)));
```

Comment:
It's not wrong, but can be simplified/optimized:
```suggestion
                        updateState(END_TIME, new DateTimeType(Instant.now().plusSeconds(duration - position)));
```

See #17725.

Reply:
> It's not wrong, but can be simplified/optimized:
> 
> See #17725.

There is one in kaleidescape also: https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.kaleidescape/src/main/java/org/openhab/binding/kaleidescape/internal/handler/KaleidescapeMessageHandler.java
Should I update that one too?

Reply:
It's very minor, so up to you if it worth to create a PR for, or just leave it for some future refactoring. 😉 The `ZonedDateTime` constructor is not planned for deprecation. In #17725 I mainly focused on removal of deprecated methods and cases where `TimeZoneProvider` could be completely eliminated because you no longer have to provide "correct" time-zone when creating a `DateTimeType`.